lib/src/main/java/io/ably/lib/transport/ConnectionManager.java

```diff
@@ -881,64 +900,74 @@ public void run() {
-                requestState(new StateIndication(ConnectionState.disconnected, disconnectError, null, null));
-                /* Start a new connection attempt. */
-                connect();
-                break;
+                case connecting:
+                    /* Close the connecting transport. */
+                    Log.v(TAG, "onAuthUpdated: closing connecting transport");
+                    ErrorInfo disconnectError = new ErrorInfo("Aborting incomplete connection with superseded auth params", 503, 80003);
+                    requestState(new StateIndication(ConnectionState.disconnected, disconnectError, null, null));
+                    /* Start a new connection attempt. */
+                    connect();
+                    break;
 
-            default:
-                /* Start a new connection attempt. */
-                connect();
-                break;
-        }
+                default:
+                    /* Start a new connection attempt. */
+                    connect();
+                    break;
+            }
 
-        if(!waitForResponse) {
-            return;
-        }
+            if(!waitForResponse) {
+                return;
+            }
 
-        /* Wait for a currentState transition into anything other than connecting or
-         * disconnected. Note that this includes the case that the connection
-         * was already connected, and the AUTH message prompted the server to
-         * send another connected message. */
-        for (;;) {
-            ErrorInfo reason = waiter.waitForChange();
-            switch (currentState.state) {
-                case connected:
-                    Log.v(TAG, "onAuthUpdated: got connected");
-                    return;
-                case connecting:
-                case disconnected:
-                    continue;
-                default:
-                    /* suspended/closed/error: throw the error. */
-                    Log.v(TAG, "onAuthUpdated: throwing exception");
-                    throw AblyException.fromErrorInfo(reason);
+            /* Wait for a currentState transition into anything other than connecting or
+             * disconnected. Note that this includes the case that the connection
+             * was already connected, and the AUTH message prompted the server to
+             * send another connected message. */
+            boolean waitingForConnected = true;
+            while (waitingForConnected) {
+                final ErrorInfo reason = waiter.waitForChange();
+                final ConnectionState connectionState = currentState.state;
+                switch (connectionState) {
+                    case connected:
+                        Log.v(TAG, "onAuthUpdated: got connected");
+                        waitingForConnected = false;
+                        break;
+
+                    case connecting:
```

Comment:
Not sure, but if it's `connecting`, it can transition to `connected` right ... that way it shouldn't break and just continue to check if the next state is connected

Reply:
Maybe we can add explicit continue ... WDYT

Reply:
I'm not quite sure what you mean, as this whole `switch` statement is validating a single state. So, `connected` is transitory and `connected` will get encountered on a subsequent iteration of the `while`.